### PR TITLE
Make use of currentColor

### PR DESCRIPTION
**Pull request by camillef**
To allow customer branding to cascade

Also includes some miscellaneous fixes picked up from reviewing the update with customer branding in Shipshape

### CODE REVIEW

File: assets/stylesheets/bootstrap/_forms.scss [hunk 1]
```diff
@@ -152,7 +152,7 @@ output {
   @include transition(border-color ease-in-out .15s);
 
   // Customize the `:focus` state to imitate native WebKit styles.
-  @include form-control-focus;
+  @include form-control-focus(currentColor);
```
Comment (camillef) [inline assets/stylesheets/bootstrap/_forms.scss:155]: I'm not sure if this works - need to test it out

File: assets/stylesheets/bootstrap/mixins/_buttons.scss [hunk 2]
```diff
@@ -3,10 +3,10 @@
 // Easily pump out default styles, as well as :hover, :focus, :active,
 // and disabled options for all buttons
 
-@mixin button-variant-plain($color, $background, $border) {
```
Comment (camillef) [inline assets/stylesheets/bootstrap/mixins/_buttons.scss:6]: The `$border` seemed to always be the same as `$color` so I've removed and used `currentColor` for all instances of both `$color` and `$border` except for the base rule `color: $color;` which will get overridden by customer branding if needed

File: assets/stylesheets/bootstrap/_button-groups.scss [hunk 3]
```diff
@@ -86,8 +86,8 @@
 .btn-group .dropdown-toggle:active,
 .btn-group.open .dropdown-toggle {
   outline: 0;
-  border-color: $brand-primary;
-  // box-shadow: 0 0 0 1px $brand-primary;
+  border-color: currentColor;
+  // box-shadow: 0 0 0 1px currentColor;
```
Comment (danielmatthew) [inline assets/stylesheets/bootstrap/_button-groups.scss:90]:
```suggestion
  // box-shadow: 0 0 0 1px currentColor;
```
```suggestion

```

**Comment by danielmatthew** [comment]
Looking good: changed the green to `rebeccapurple` and button borders followed suit…

Screenshot 2020-09-24 at 08 58 38

…but: what about that pesky btn-primary state, and inputs, with a `color` of `#ffffff` and `#000000` respectively 🤔

**Comment by danielmatthew** [comment]
Do you think there's anything we can do with https://github.com/talis/rl-app/blob/master/src/classes/api/BrandingAPI.class.php to dump out the two core customer colours in the following format? 
Your instances of `currentColor` can be swapped for the appropriate variable.
```html
<style>
--customerPrimary: #101;
--customSecondary: #999;
</style>
```

**Comment by camillef** [comment]
> Do you think there's anything we can do with https://github.com/talis/rl-app/blob/master/src/classes/api/BrandingAPI.class.php to dump out the two core customer colours in the following format?
>
> ```
> <style>
> :root {
>   --customerPrimary: #101;
>   --customSecondary: #999;
> }
> </style>
> ```

I don't see why not! Might we also need the navbar colour? I'm not sure what the plan is for focus on that...

**Comment by danielmatthew** [comment]
> I don't see why not! Might we also need the navbar colour? I'm not sure what the plan is for focus on that...

Ah, blummin' heck. Yes, that would be another property we'd need. Focus/active state-wise, the outline should be left at our  be our default white state for now.

**Comment by camillef** [comment]
> Ah, blummin' heck. Yes, that would be another property we'd need. Focus/active state-wise, the outline should be left at our be our default white state for now.

Raised a PR: https://github.com/talis/rl-app/pull/2808

File: assets/stylesheets/bootstrap/_buttons.scss [hunk 4]
```diff
@@ -96,8 +96,8 @@ a.btn {
 
 // Make a button look and behave like a link
 .btn-link {
-  font-weight: 400;
```
Comment (camillef) [inline assets/stylesheets/bootstrap/_buttons.scss:99]: Redefined two lines below

File: assets/stylesheets/bootstrap/_buttons.scss [hunk 5]
```diff
@@ -96,8 +96,8 @@ a.btn {
...
   color: $link-color;
+  border-color: transparent;
```
Comment (camillef) [inline assets/stylesheets/bootstrap/_buttons.scss:100]: Ensure `btn-link` always has a transparent border, even if it is also a `btn-danger`

Examples are 
- the cancel button on the action bar when item has been cut
image
- the set importance button when no importance has been set
image

File: assets/stylesheets/bootstrap/mixins/_alerts.scss [hunk 6]
```diff
@@ -15,4 +15,10 @@
     color: $text-color !important;
     text-decoration: underline;
   }
+  .btn-alert {
+    text-decoration: none;
+    &:hover {
+      color: $white !important;
+    }
+  }
```
Comment (camillef) [inline assets/stylesheets/bootstrap/mixins/_alerts.scss:23]: Overwrite the two rules immediately above when we have an `a` styled as a button inside an alert using `btn-alert`

Should look like 
image
and on hover
image

File: assets/stylesheets/bootstrap/mixins/_buttons.scss [hunk 7]
```diff
@@ -199,6 +199,10 @@
     background-image: none;
   }
 
+  &:hover:active {
+    background-color: $black;
```
Comment (camillef) [inline assets/stylesheets/bootstrap/mixins/_buttons.scss:203]: Should favour hover `background-color` when both `hover` and `active`

**Comment by camillef** [comment]
> Your instances of `currentColor` can be swapped for the appropriate variable.

Having experimented with CSS variables a little, they really don't play nicely with SCSS... as soon as you want to pass the variable to a function, the sass compiler errors with messages along the lines of `argument `$color` of `darken($color, $amount)` must be a color`

**Comment by danielmatthew** [comment]
> Having experimented with CSS variables a little, they really don't play nicely with SCSS...

That's OK. A custom property could be modified with a built-in CSS function instead. We'd modify the `l` of a `hsla()` colour to darken it by the required amount.

File: assets/stylesheets/bootstrap/mixins/_buttons.scss (hunk 7 above)
Comment (danielmatthew) [reply assets/stylesheets/bootstrap/mixins/_buttons.scss:203]: Was it going transparent again when clicked?

Comment (camillef) [reply assets/stylesheets/bootstrap/mixins/_buttons.scss:203]: Yep
Oct-05-2020 11-23-41

File: dist/app.css [hunk 8]
```diff
@@ -226,7 +226,7 @@ a {
     text-decoration: none; }
```
Comment (danielmatthew) [inline dist/app.css:226]: Don't worry about commiting changes in `app.css`: there's a task to generate the dist files when the version is bumped.

**Review by danielmatthew — APPROVED** [review]
Happy with how it's rendering on http://localhost:8000/ ?

File: dist/app.css (hunk 8 above)
Comment (danielmatthew) [reply dist/app.css:226]: Oh yeah… the release process. I've been using conventional-commits to help with this. Would you agree that this commit is probably of 'fix', so will bump the patch version?

Comment (camillef) [reply dist/app.css]: Yep, agree

**Pull request merged**